File: app/src/main/java/com/chooloo/www/callmanager/ui/call/CallMvpPresenter.java
package com.chooloo.www.callmanager.ui.call;

import android.view.KeyEvent;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface CallMvpPresenter<V extends CallMvpView> extends MvpPresenter<V> {

    boolean onBackPressed();

    void onAnswerClick();

    void onRejectClick();

    void onAddCallClick();

    void onKeypadClick();

    void onMuteClick(boolean isActivated);

    void onSpeakerClick(boolean isActivated);

    void onHoldClick(boolean isActivated);

    void onDialpadKeyClick(int keyCode, KeyEvent event);

    void onDetailsChanged();

    void onStateChanged();
}

File: app/src/main/java/com/chooloo/www/koler/ui/call/CallMvpPresenter.kt
package com.chooloo.www.koler.ui.call

import android.telecom.Call
import com.chooloo.www.koler.ui.base.MvpPresenter

interface CallMvpPresenter<V : CallMvpView> : MvpPresenter<V> {
    fun onAnswerClick()
    fun onRejectClick()
    fun onInitialUI()
    fun onDetailsChanged(details: Call.Details?)
    fun onStateChanged(state: Int?)
}